gandalf/src/main/java/io/github/btkelly/gandalf/checker/HistoryChecker.java

```diff
@@ -29,13 +31,27 @@
      * @param alert item to compare against the history
      * @return {@code true} if provided alert matches the history
      */
-    boolean contains(Alert alert);
+    boolean contains(@NonNull final Alert alert);
+
+    /**
+     * Saves the provided {@link OptionalUpdate} to shared preferences.
```

Comment:
Perhaps good context to add:

- why do we do this? how is the saved `OptionalUpdate` used in the future? Will this overwrite any existing saved `OptionalUpdate`?